### PR TITLE
gui: fixes to segfaults and allow for better debugging

### PR DESCRIPTION
**Pull request by gadfort**
Adds:
- tcl commands to access heatmap settings
- avoids using corner in dbSta for power density as this can segfault
- ensure nodes are saved after failed IRDrop analysis for better debugging.

### CODE REVIEW

**Comment by maliberty** [comment]
> avoids using corner in dbSta for power density as this can segfault
What is the issue?  Is this something that needs fixing?

File: src/psm/src/pdnsim.cpp [hunk 1]
```diff
@@ -135,10 +163,10 @@ void PDNSim::analyzePowerGrid(const std::string& voltage_file,
   auto irsolve_h = getIRSolver(enable_em);
 
   if (!irsolve_h->build(error_file)) {
+    saveIRDrop(irsolve_h.get());
     logger_->error(
         utl::PSM, 78, "IR drop setup failed.  Analysis can't proceed.");
```
Comment (maliberty) [inline src/psm/src/pdnsim.cpp:168]: Why do we save the results on failure?

**Comment by gadfort** [comment]
@maliberty if the gui is opened and the `corner_` gets set, but then a new set of corners are defined, the `corner_` is destroyed by OpenSTA, but this reference doesn't know about it causing it to segfault when attempting to run.

File: src/psm/src/pdnsim.cpp (hunk 1 above)
Comment (gadfort) [reply src/psm/src/pdnsim.cpp:168]: the GUI debugger relies on this to be saved. At present, we can only debug PSM when it is successful. This ensures the information about the nodes is copied over so we can debug it when it "claims" the PDN is not connected

**Review by maliberty — APPROVED** [review]

**Pull request merged**